cmd/api/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+)
+
+var Version string
+
+type config struct {
+	port int
+	env  string
+}
+type application struct {
+	config config
+	// TODO: Add logger
+}
+
+var cfg config
+
+// Read value of command-line flags
+func init() {
+	flag.IntVar(&cfg.port, "port", 8080, "API server port")
+	flag.StringVar(&cfg.env, "env", "development", "Environment (development|staging|production)")
+	flag.Parse()
+}
+
+func main() {
+	mux := http.NewServeMux()
```

Comment:
I prefer to have an interface which returns the server object. But we can refactor it later. main.go should as little as possible